dev/cognitiveservices/data-plane/Language/analyzetext.json

```diff
@@ -1743,6 +1743,664 @@
+        }
+      },
+      "required": [
+        "unit"
+      ]
+    },
+    "TemperatureResolution": {
+      "description": "Represents the temperature entity resolution model.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseResolution"
+        },
+        {
+          "$ref": "#/definitions/QuantityResolution"
+        }
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "TemperatureResolution",
+      "properties": {
+        "unit": {
+          "type": "string",
+          "enum": [
+            "Unspecified",
+            "Fahrenheit",
+            "Kelvin",
+            "Rankine",
+            "Celsius"
+          ],
+          "x-ms-enum": {
+            "name": "TemperatureUnit",
+            "modelAsString": true
+          },
+          "description": "The temperature Unit of measurement."
+        }
+      },
+      "required": [
+        "unit"
+      ]
+    },
+    "WeightResolution": {
+      "description": "Represents the weight entity resolution model.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseResolution"
+        },
+        {
+          "$ref": "#/definitions/QuantityResolution"
+        }
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "WeightResolution",
+      "properties": {
+        "unit": {
+          "type": "string",
+          "enum": [
+            "Unspecified",
+            "Kilogram",
+            "Gram",
+            "Milligram",
+            "Gallon",
+            "MetricTon",
+            "Ton",
+            "Pound",
+            "Ounce",
+            "Grain",
+            "PennyWeight",
+            "LongTonBritish",
+            "ShortTonUS",
+            "ShortHundredWeightUS",
+            "Stone",
+            "Dram"
+          ],
+          "x-ms-enum": {
+            "name": "WeightUnit",
+            "modelAsString": true
+          },
+          "description": "The weight Unit of measurement."
+        }
+      },
+      "required": [
+        "unit"
+      ]
+    },
+    "CurrencyResolution": {
+      "description": "Represents the currency entity resolution model.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BaseResolution"
+        },
+        {
+          "$ref": "#/definitions/QuantityResolution"
+        }
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "CurrencyResolution",
+      "properties": {
+        "iso4217": {
+          "type": "string",
+          "description": "The alphabetic code based on another ISO standard, ISO 3166, which lists the codes for country names. The first two letters of the ISO 4217 three-letter code are the same as the code for the country name, and, where possible, the third letter corresponds to the first letter of the currency name."
+        },
+        "value": {
+          "type": "number",
+          "format": "double",
+          "description": "The money amount captured in the extracted entity"
+        },
+        "unit": {
```

Comment:
It might be better to add an extensible enum type for all of the currencies. But feel free to add it in next version.

Reply:
Let's add this in the next version.